datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedPartitionAssigner.java

```diff
@@ -91,7 +91,7 @@ public Map<String, Set<DatastreamTask>> assignPartitions(
         tasks.forEach(task -> {
           if (task.getTaskPrefix().equals(datastreamGroupName)) {
             Set<String> retainedPartitions = new HashSet<>(task.getPartitionsV2());
-            retainedPartitions.retainAll(new HashSet<>(partitionMetadata.getPartitions()));
+            retainedPartitions.retainAll(partitionMetadata.getPartitions());
```

Comment:
Is partitionMetadata.getPartitions() used only in the below three places ?

Reply:
Its used in other places as well but some basic operations on list and set are common, hence no change needed there.